test/type-provider.test.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('..')
+
+test('Should export withTypeProvider function', t => {
+  t.plan(1)
+  try {
+    Fastify().withTypeProvider()
+    t.pass()
+  } catch (e) {
+    t.fail()
+  }
+})
+
+test('Should return same instance', t => {
+  t.plan(1)
+  const fastify = Fastify()
+  const next = fastify.withTypeProvider()
+  if (next === fastify) {
```

Comment:
This is the node tap way.
```suggestion
  t.equal(next,fastify)
```

Reply:
It seems like suggestion was applied and now we have syntax error in this file 😅